app/views/customers/_form.html.erb

```diff
@@ -11,27 +11,51 @@
     </div>
   <% end %>
 
-  <div class="field">
-    <%= form.label :first_name %>
-    <%= form.text_field :first_name %>
-  </div>
+  <% if current_page?(new_customer_path) %>
+    <div class="row">
+      <div class="form-group col-6 col-md">
+  <% else %>
+    <div class="card-body">
+      <div class="form-group">
+  <% end %>
+      <%= form.label :first_name %>
+      <%= form.text_field :first_name, class: "form-control form-control-sm", autofocus: true %>
+    </div>
 
-  <div class="field">
-    <%= form.label :last_name %>
-    <%= form.text_field :last_name %>
-  </div>
+    <% if current_page?(new_customer_path) %>
+      <div class="form-group col-6 col-md">
+    <% else %>
+      <div class="form-group">
+    <% end %>
+      <%= form.label :last_name %>
+      <%= form.text_field :last_name, class: "form-control form-control-sm" %>
+    </div>
 
-  <div class="field">
-    <%= form.label :phone %>
-    <%= form.text_field :phone %>
-  </div>
+    <% if current_page?(new_customer_path) %>
+      <div class="form-group col-6 col-md">
+    <% else %>
+      <div class="form-group">
+    <% end %>
+      <%= form.label :phone %>
+      <%= form.text_field :phone, class: "form-control form-control-sm", placeholder: "10 digits..." %>
+    </div>
 
-  <div class="field">
-    <%= form.label :email %>
-    <%= form.text_field :email %>
+    <% if current_page?(new_customer_path) %>
+      <div class="form-group col-6 col-md">
+    <% else %>
+      <div class="form-group">
+    <% end %>
+      <%= form.label :email %>
+      <%= form.text_field :email, class: "form-control form-control-sm", placeholder: "name@example.com" %>
+    </div>
   </div>
 
-  <div class="actions">
-    <%= form.submit %>
+  <div class="mb-3">
+    <%= form.submit class: 'btn btn-outline-primary btn-sm' %>
+    <% if current_page?(new_customer_path) %>
+      <%= link_to 'Cancel', customers_path, class: 'btn btn-outline-primary btn-sm', role: 'button' %>
+    <% else %>
+      <%= link_to 'Cancel', customer_path(customer), class: 'btn btn-outline-primary btn-sm', role: 'button' %>
+    <% end %>
```

Comment:
This is the reason the `./spec/requests/customers_spec.rb` test is failing. In the controller you are doing `render :new` when there is an error so it seems like `current_page?(new_customer_path)` should return `true`. I put a debugger here and it is returning false after the `render :new`. I'm honestly not sure why it isn't working, but this is where the error for the test is. Because that is returning `false`, it is rendering the 2nd `link_to` that you have, which is expecting a `customer` to be set. That is why it is throwing an error for a "show" template, because the `customer` doesn't exist.